Enable `e2eRealApiOrderDetails` UI test

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
`OrdersRealAPi.e2eRealApiOrderDetails()` was ignored before since it was failing. This test was fixed in https://github.com/woocommerce/woocommerce-android/pull/12963 and https://github.com/woocommerce/woocommerce-android/pull/12906 before and this PR removes `Ignore` annotation from it. I also removed `@Retry(numberOfTimes = 1)` from it since it's not flaky.

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
Run `OrdersRealAPi.e2eRealApiOrderDetails()` a couple times ti ensure it's passing every time.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Ran `OrdersRealAPi.e2eRealApiOrderDetails()` a couple of times.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Remove ignoring e2eRealApiOrderDetails

irfano committed:
Remove retrying from e2eRealApiOrderDetails()

hafizrahman reviewed (commented):
I manually re-run the test five times both on a phone simulator API 35, and they all pass fine.

However on a tablet simulator (Pixel tablet API 32) it keeps failing. I'm sending you the recording in Slack p1733375323312509-slack-C03L1NF1EA3




**Aside:**
I was curious if there's a way to do this repeat tests automatically, because in Xcode there is such an option:

![image](https://github.com/user-attachments/assets/a53eb2ea-d916-4109-8d76-4dfd8c439609)
([source](https://blog.lars-richter.dev/quick-tip-running-tests-repeatedly-in-xcode))

I found an option in Android Studio but its only for JUnit tests:
![Screenshot 2024-12-05 at 11 53 50](https://github.com/user-attachments/assets/94f60a38-cd19-4720-a507-9676832d2544)

irfano committed:
Remove clearing the search bar from OrdersRealAPI's After function

We only fill the search bar in the e2eRealApiOrdersSearch() function, and it already clears the search bar after filling it. Therefore, it is unnecessary to clear it again in the After function.

irfano committed:
Fix assertOrderId() function for tablets

irfano committed:
Fix goBackToOrdersScreen() function for tablets

irfano commented:
Good catch on the tablet issue, Hafiz! I've implemented some fixes:

- Removed clearing search bar in the `After` function since it was unnecessary and was causing failures on tablets. https://github.com/woocommerce/woocommerce-android/pull/13066/commits/374bd26b1f02278cf5faf1d51bc67f753e88e79f
- Updated `assertOrderId()`. Previously, it relied on `R.id.toolbar`, but on tablets, two `R.id.toolbar` instances were visible. I've updated it to use a more specific matcher to identify the detail screen's toolbar. f64820cea3676c3831dfe714d8d5f18560473ae7
- Updated `goBackToOrdersScreen()`.  If the order list is already open, it won't press back. Because on tablets, the order list is displayed alongside the detail screen. ac5537ed6e7b3e15424156e6602488a64e227b53

Please also test `ScreenshotTest` and `OrdersUITest` to ensure these changes don't break them. Note that these tests might still fail on tablets, but they are already failing on the trunk. Fixing those failures is outside the scope of this PR.

> I was curious if there's a way to do this repeat tests automatically

I’m not sure if it’s possible; I always run it manually multiple times.

hafizrahman reviewed (approved):
Thanks for the update. Tested the new update a few times in tablet and they're running nicely.

> Please also test ScreenshotTest and OrdersUITest to ensure these changes don't break them. Note that these tests might still fail on tablets, but they are already failing on the trunk. Fixing those failures is outside the scope of this PR.

Got it. I tested these on phones and they're going fine, I believe that's OK for now.
